wordpress-mobile/WordPress-iOS
# Issues/11078 video tag sizing

Fixes #11078 

This PR should resolve a sizing glitch with certain video tags missing width and height attributes.  Additionally it improves and simplifies some sizing logic around embeds, moves scripts into external flies for improved readability, and relies on a JS bridge to notify the embed class when its webview's geometry is known. 

To recap: the sizing issue occurred when a video tag did not have width or height attributes.  In this scenario its aspect ratio is not known and we have to wait until the web view finishes loading its content in order to get its internal page geometry.  The trick is that video tags only retrieve meta data about their content *after* the web page has loaded.  A video's meta data includes width and height, and a web page's size is adjusted once these values are known.  Before this patch the video embeds and their text attachments never received notice of the updated geometry leading to cropping if the original bounding box was too small. 

The fix relies on improved javascript that listens for when a page is loaded.  If the page contains a video tag, it listens for when that tag's meta data is updated. Relevant sizing information is sent to an observer and processed to ensure the video's correct aspect ratio is respected.

To test:
View posts with video embeds.  Ensure the embedded content is properly sized and displayed.
I tested with these two:
https://tame-sacred.jurassic.ninja/2019/02/18/videopress-block-test/
https://aerychtest.wordpress.com/2014/10/02/embedables/

@jkmassel could I trouble you for a review since you're already familiar with the issue?
